include/zephyr/bluetooth/mesh/health_cli.h

```diff
@@ -99,6 +99,18 @@ struct bt_mesh_health_cli {
 			       uint8_t test_id, uint16_t cid, uint8_t *faults,
 			       size_t fault_count);
 
+	/** @brief Optional callback for updating the message to be sent as periodic publication.
+	 *
+	 *  This callback is called before sending the periodic publication message.
+	 *  The callback can be used to update the message to be sent.
```

Comment:
We need to say also that if the callback is not implemented, the periodic publication can still be enabled, but nothing will be sent.